LEGO1/lego/legoomni/src/actors/act2actor.cpp

```diff
@@ -115,11 +138,197 @@ MxResult Act2Actor::VTable0x9c()
 	}
 }
 
-// STUB: LEGO1 0x10018c30
-// STUB: BETA10 0x1000cb52
+// FUNCTION: LEGO1 0x10018c30
+// FUNCTION: BETA10 0x1000cb52
 void Act2Actor::VTable0x70(float p_time)
 {
-	// TODO
+	int dummy1; // for BETA10, not sure what it is being used for
+	ViewManager* vm;
+	LegoROI* roiPepper;
```

Comment:
Moving the declarations of all those variables to where they are used seems to improve the match to about ~78% for me - it still retains the stack allocation issue (and the stack size mismatch) but the diff becomes much more readable and it seems reasonable that at least some of those variables were not declared at the beginning (I'd say at least `childPosition` to `distance3`, but maybe also the ROIs)

Reply:
Interesting - the exact reverse happened when I had `distance3` instead of `p_time` in the other place. I changed it back.